src/main/java/ua/danit/final_project/controllers/VacancyController.java

```diff
@@ -13,87 +12,40 @@
 public class VacancyController extends SessionAware {
 
   private final VacancyServiceCrud vacancyService;
-  private final PositionService positionService;
 
   @Autowired
-  public VacancyController(VacancyServiceCrud vacancyService, PositionService positionService) {
+  public VacancyController(VacancyServiceCrud vacancyService) {
     this.vacancyService = vacancyService;
-    this.positionService = positionService;
   }
 
   @GetMapping
   public List<VacancyDto> getVacanciesDto() {
-    return vacancyService.getOpenVacancies()
-            .stream()
-            .map(VacancyDto::new)
-            .collect(Collectors.toList());
-  }
-
-  @GetMapping("/{id}")
-  public ResponseEntity<VacancyDto> getVacancyById(@PathVariable("id") Long id) {
-    return ResponseEntity.ok(new VacancyDto(vacancyService.getById(id)));
+    return vacancyService.getOpenVacancies();
   }
 
   @PostMapping
-  public ResponseEntity<VacancyDto> createVacancy(@RequestBody VacancyDto vacancyDto) {
-    User userFromToken = getCurrentUser();
-
-    Vacancy vacancy = new Vacancy();
-
-    vacancy.setVacancyStatus(Vacancy.VacancyStatus.OPENED);
-    vacancy.setUser(userFromToken);
-    vacancy.setPosition(positionService.getById(vacancyDto.getPositionId()));
-    vacancy.setSalary(vacancyDto.getSalary());
-    vacancy.setInfo(vacancyDto.getInfo());
-    vacancy.setPublication(vacancyDto.getPublication());
-
-    return ResponseEntity.ok(new VacancyDto(vacancyService.save(vacancy)));
+  public VacancyDto createVacancy(@RequestBody VacancyDto vacancyDto) {
+    return vacancyService.create(vacancyDto);
   }
 
   @PutMapping
-  public ResponseEntity<VacancyDto> updateVacancy(@RequestBody VacancyDto vacancyDto) {
-    User userFromToken = getCurrentUser();
-
-    Vacancy vacancy = new Vacancy();
-
-    vacancy.setId(vacancyDto.getId());
-    vacancy.setUser(userFromToken);
-    vacancy.setPosition(positionService.getById(vacancyDto.getPositionId()));
-    vacancy.setSalary(vacancyDto.getSalary());
-    vacancy.setVacancyStatus(vacancyDto.getStatus());
-    vacancy.setInfo(vacancyDto.getInfo());
-    vacancy.setPublication(vacancyDto.getPublication());
-
-    try {
-      return ResponseEntity.ok(new VacancyDto(vacancyService.updateVacancy(vacancy, userFromToken)));
-    } catch (IllegalAccessException e) {
-      return  ResponseEntity.status(HttpStatus.FORBIDDEN).build();
-    }
+  public VacancyDto updateVacancy(@RequestBody VacancyDto vacancyDto) {
+    return vacancyService.updateVacancy(vacancyDto);
   }
 
   @DeleteMapping("/{id}")
-  public ResponseEntity deleteVacancy(@PathVariable("id") Vacancy vacancy) {
-    User userFromToken = getCurrentUser();
-
-    try {
-      vacancyService.deleteVacancy(vacancy, userFromToken);
-      return ResponseEntity.ok().build();
-    } catch (IllegalAccessException e) {
-      return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
-    }
-
+  public ResponseEntity<?> deleteVacancy(@PathVariable("id") Vacancy vacancy) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove usage of generic wildcard type. ([squid:S1452](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AS1452))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=149&issues=AWaRCmsFsSPV6ZXI9BXG&open=AWaRCmsFsSPV6ZXI9BXG)

[ISSUE_KEY]: AWaRCmsFsSPV6ZXI9BXG